task/host-sp-comms/src/main.rs

```diff
@@ -890,13 +964,70 @@ impl ServerImpl {
                 );
                 response_len
             }
+            Key::EtcSystem => {
+                let response_len = self.host_kv_storage.etc_system_len;
+                if response_len == 0 {
+                    return Err(KeyLookupResult::NoValueForKey);
+                }
+
+                self.tx_buf.encode_response(
+                    sequence,
+                    &SpToHost::KeyLookupResult(KeyLookupResult::Ok),
+                    |buf| {
+                        // Statically guarantee we have sufficient space in
+                        // `buf` for longest possible ETC_SYSTEM blob.
+                        const_assert!(
+                            MIN_SP_TO_HOST_FILL_DATA_LEN >= MAX_ETC_SYSTEM_LEN
+                        );
+                        buf[..response_len].copy_from_slice(
+                            &self.host_kv_storage.etc_system[..response_len],
+                        );
+                        response_len
+                    },
+                );
+                response_len
+            }
+            Key::DtraceConf => {
+                let response_len = self.host_kv_storage.dtrace_conf_len;
+                if response_len == 0 {
+                    return Err(KeyLookupResult::NoValueForKey);
+                }
+
+                self.tx_buf.encode_response(
+                    sequence,
+                    &SpToHost::KeyLookupResult(KeyLookupResult::Ok),
+                    |buf| {
+                        // `MIN_SP_TO_HOST_FILL_DATA_LEN` is calculated assuming
+                        // `SpToHost::MAX_SIZE`, but we know in this callback
+                        // we're appending to
+                        // `SpToHost::KeyLookupResult(KeyLookupResult::Ok)`,
+                        // which is only 2 bytes. Recompute the exact max space
+                        // we have for our response, then statically guarantee
+                        // we have sufficient space in `buf` for longest
+                        // possible DTRACE_CONF blob.
+                        const SP_TO_HOST_FILL_DATA_LEN: usize =
+                            MIN_SP_TO_HOST_FILL_DATA_LEN + SpToHost::MAX_SIZE
+                                - 2;
+                        const_assert!(
+                            SP_TO_HOST_FILL_DATA_LEN >= MAX_DTRACE_CONF_LEN
+                        );
```

Comment:
Should the `Key::EtcSystem` block also do this calculation, instead of simply checking `MIN_SP_TO_HOST_FILL_DATA_LEN >= MAX_ETC_SYSTEM_LEN`?

Reply:
It could but doesn't need to - `MIN_SP_TO_HOST_FILL_DATA_LEN` is the minimum space available; i.e., even with the largest possible `SpToHost` variant. That minimum does _not_ leave room for `MAX_DTRACE_CONF_LEN`, so we had to do the more accurate calculation here to show that we really do (statically) have enough space, even though `MAX_DTRACE_CONF_LEN > MIN_SP_TO_HOST_FILL_DATA_LEN`.

Reply:
Gotcha, that makes sense